File: src/main/java/com/kevin/data/service/CounterService.java
package com.kevin.data.service;

import com.kevin.data.dao.CounterDAO;
import com.kevin.data.model.Counter;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class CounterService {
    private int id = 1;
    @Autowired
    private CounterDAO counterDAO;

    public int getCounter() {
        Counter c = counterDAO.findById(id).orElseGet(() -> {
            Counter nc = new Counter();
            nc.setCount(0);
            nc.setId(id);
            return nc;
        });
        return c.getCount();
    }

    public int upsertCount() {
        Counter c = counterDAO.findById(id).orElseGet(() -> {
            Counter nc = new Counter();
            nc.setCount(0);
            nc.setId(id);
            return nc;
        });
        c.setCount(c.getCount() + 1);
        counterDAO.save(c);
        return c.getCount();
    }

    public int clearCount() {
        Counter c = counterDAO.findById(id).orElseGet(() -> {
            Counter nc = new Counter();
            nc.setCount(0);
            nc.setId(id);
            return nc;
        });
        c.setCount(0);
        counterDAO.save(c);
        return c.getCount();
    }
}

File: src/main/kotlin/com/kevin/data/service/CounterService.kt
package com.kevin.data.service

import com.kevin.data.dal.dao.CounterDAO
import com.kevin.data.dal.entity.Counter
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service


@Service
class CounterService {
    private val id = 1

    @Autowired
    private lateinit var counterDAO: CounterDAO

    fun getCount():Int{
        val cnt = counterDAO.findById(id).orElse(Counter(id=1,count=0))
        counterDAO.save(cnt!!)
        return cnt.count
    }
    fun upsertCount(): Int {
        val cnt = counterDAO.findById(id).orElse(Counter(id=1,count=0))!!
        cnt.count = cnt.count+1
        counterDAO.save(cnt)
        return cnt.count
    }

    fun clearCount(): Int {
        val cnt = counterDAO.findById(id).orElse(Counter(id=1,count=0))!!
        cnt.count = 0
        counterDAO.save(cnt)
        return cnt.count
    }
}

